From c4b06654a7c35ce93b342d9efffb45baeb8c5078 Mon Sep 17 00:00:00 2001
From: lijuang <quic_lijuang@quicinc.com>
Date: Thu, 26 Sep 2024 18:17:12 +0800
Subject: [PATCH 55/66] arm64: dts: qcom: qcs615: add ADSP and CDSP nodes

Add nodes for remoteprocs: ADSP and CDSP for QCS615 SoC.

Signed-off-by: Lijuan Gao <quic_lijuang@quicinc.com>
---
 arch/arm64/boot/dts/qcom/qcs615.dtsi | 96 ++++++++++++++++++++++++++++
 1 file changed, 96 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/qcs615.dtsi b/arch/arm64/boot/dts/qcom/qcs615.dtsi
index f8b0a789fba5..441bd9d8836c 100644
--- a/arch/arm64/boot/dts/qcom/qcs615.dtsi
+++ b/arch/arm64/boot/dts/qcom/qcs615.dtsi
@@ -412,6 +412,16 @@ smem_region: smem@86000000 {
 			no-map;
 			hwlocks = <&tcsr_mutex 3>;
 		};
+
+		rproc_cdsp_mem: rproc-cdsp@93b00000 {
+			reg = <0x0 0x93b00000 0x0 0x1e00000>;
+			no-map;
+		};
+
+		rproc_adsp_mem: rproc-adsp@95900000 {
+			reg = <0x0 0x95900000 0x0 0x1e00000>;
+			no-map;
+		};
 	};
 
 	soc: soc@0 {
@@ -1228,6 +1238,47 @@ adreno_smmu: iommu@50a0000 {
 			             <GIC_SPI 597 IRQ_TYPE_LEVEL_HIGH>;
 		};
 
+		remoteproc_cdsp: remoteproc-cdsp@8300000 {
+			compatible = "qcom,qcs615-cdsp-pas";
+			reg = <0x0 0x8300000 0x0 0x100000>;
+
+			interrupts-extended = <&intc GIC_SPI 578 IRQ_TYPE_EDGE_RISING>,
+			                      <&cdsp_smp2p_in 0 IRQ_TYPE_EDGE_RISING>,
+			                      <&cdsp_smp2p_in 2 IRQ_TYPE_EDGE_RISING>,
+			                      <&cdsp_smp2p_in 1 IRQ_TYPE_EDGE_RISING>,
+			                      <&cdsp_smp2p_in 3 IRQ_TYPE_EDGE_RISING>;
+			interrupt-names = "wdog",
+			                  "fatal",
+			                  "handover",
+			                  "ready",
+			                  "stop-ack";
+
+			clocks = <&rpmhcc RPMH_CXO_CLK>;
+			clock-names = "xo";
+
+			power-domains = <&rpmhpd RPMHPD_CX>;
+			power-domain-names = "cx";
+
+			interconnects = <&aggre1_noc MASTER_CRYPTO 0 &mc_virt SLAVE_EBI1 0>;
+			interconnect-names = "crypto_ddr";
+
+			memory-region = <&rproc_cdsp_mem>;
+
+			qcom,qmp = <&aoss_qmp>;
+
+			status = "disabled";
+
+			qcom,smem-states = <&cdsp_smp2p_out 0>;
+			qcom,smem-state-names = "stop";
+
+			glink-edge {
+				interrupts = <GIC_SPI 574 IRQ_TYPE_EDGE_RISING>;
+				label = "cdsp";
+				qcom,remote-pid = <5>;
+				mboxes = <&apss_shared 4>;
+			};
+		};
+
 		smem: qcom,smem@8600000 {
 			compatible = "qcom,smem";
 			memory-region = <&smem_region>;
@@ -1592,6 +1643,13 @@ aoss_qmp: power-controller@c300000 {
 			#power-domain-cells = <1>;
 		};
 
+		qmp_aop: qcom,qmp-aop@c300000 {
+			compatible = "qcom,qmp-mbox";
+			qcom,qmp = <&aoss_qmp>;
+			label = "aop";
+			#mbox-cells = <1>;
+		};
+
 		sram@c3f0000 {
 			compatible = "qcom,rpmh-stats";
 			reg = <0x0 0x0c3f0000 0x0 0x400>;
@@ -1931,6 +1989,44 @@ rpmhpd_opp_turbo_l1: opp-9 {
 				};
 			};
 		};
+
+		remoteproc_adsp: remoteproc-adsp@62400000 {
+			compatible = "qcom,qcs615-adsp-pas";
+			reg = <0x0 0x62400000 0x0 0x00100>;
+
+			interrupts-extended = <&intc GIC_SPI 162 IRQ_TYPE_EDGE_RISING>,
+			                      <&adsp_smp2p_in 0 IRQ_TYPE_EDGE_RISING>,
+			                      <&adsp_smp2p_in 2 IRQ_TYPE_EDGE_RISING>,
+			                      <&adsp_smp2p_in 1 IRQ_TYPE_EDGE_RISING>,
+			                      <&adsp_smp2p_in 3 IRQ_TYPE_EDGE_RISING>;
+			interrupt-names = "wdog", "fatal", "handover",
+					  "ready", "stop-ack";
+
+			clocks = <&rpmhcc RPMH_CXO_CLK>;
+			clock-names = "xo";
+
+			power-domains = <&rpmhpd RPMHPD_CX>;
+			power-domain-names = "cx";
+
+			memory-region = <&rproc_adsp_mem>;
+
+			mboxes = <&qmp_aop 0>;
+
+			interconnects = <&aggre1_noc MASTER_CRYPTO 0 &mc_virt SLAVE_EBI1 0>;
+			interconnect-names = "crypto_ddr";
+
+			qcom,smem-states = <&adsp_smp2p_out 0>;
+			qcom,smem-state-names = "stop";
+
+			status = "disabled";
+
+			glink_edge: glink-edge {
+				interrupts = <GIC_SPI 170 IRQ_TYPE_EDGE_RISING>;
+				label = "adsp";
+				qcom,remote-pid = <2>;
+				mboxes = <&apss_shared 24>;
+			};
+		};
 	};
 
 	arch_timer: timer {
-- 
2.34.1

